ServerCore/Pages/Puzzles/Index.cshtml.cs

```diff
@@ -40,9 +40,8 @@ public async Task<IActionResult> OnGetAsync()
             }
 
             Puzzles = await (from Puzzle p in query
-                             join ContentFile joinFile in _context.ContentFiles on p equals joinFile.Puzzle into fileJoin
+                             join ContentFile joinFile in _context.ContentFiles.Where((j) => j.FileType == ContentFileType.Puzzle) on p equals joinFile.Puzzle into fileJoin
```

Comment:
Mixing the two kinds of LINQ syntax is a bit weird. I'm not certain of the right syntax, so you can leave this as-is if you need to. My guess on the right syntax would be:

```
from Puzzle p in query
let puzzleFiles = from content in _context.ContentFiles
where content.FileType == ContentFileType.Puzzle
join ContentFile joinFile in puzzleFiles on p equals joinFile.Puzzle into fileJoin
from ContentFile file in fileJoin.DefaultIfEmpty()
orderby p.Group, p.OrderInGroup, p.Name
select new PuzzleView { Puzzle = p, Content = file }).ToListAsync();
```